app/models/user.rb

```diff
@@ -32,11 +35,36 @@ def self.search(input)
     all
   end
 
+  # Verifies the token existence, fetches its user and validates its expiration
+  # and invalidates the user token if expired.
+
+  def self.verify_reset_token!(token)
+    digest = generate_digest(token)
+
+    user = find_by reset_digest: digest
+    return unless user
+
+    return unless user.reset_sent_at
+
+    expired = reset_token_expired?(user.reset_sent_at)
+
+    user.invalidate_reset_token! if expired
+
+    return if expired
+
+    user
```

Comment:
This returns `nil` and `user` not `true` and `false`

Reply:
You prefer it to be an explicit `true|false`?

Reply:
This method will:
1. Generate a digest for the given the token.
2. Find a user with the corresponding digest.
3. Validate the expiration of the token.
For valid tokens the method should return the found user that's why we need it to return its record for valid tokens.
Making the method return an explicit `true` value will cause to do an extra step on the action:
1. Re-generate a digest for the given token.
2. Re-find the user by digest.
Which feels unnecessary since in the validation process it requires generating the digest and finding the user record which is guaranteed to be truthy.